### PR TITLE
pkg/lwip: add support for esp32 Ethernet device

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR adds the support for ESP32 Ethernet device to `pkg/lwip` in same way as PR #12895 does. Addititionally, it adds the IPv4 support for the ESP32 Ethernet device as PR #12903.

### Testing procedure

Flash `tests/lwip` with enabled module `esp_wifi` on any ESP32 board with an Ethernet interface (for example Olimex ESP32-EVB)
```
USEMODULE=esp_eth make BOARD=esp32-olimex-evb -C tests/lwip flash term
```
and execute `ifconfig` on ESP32 node and ping the node from any machine in the LAN.

### Issues/PRs references

Related to PR #12903.

### CODE REVIEW

File: pkg/lwip/contrib/lwip.c [hunk 1]
```diff
@@ -145,6 +158,21 @@ void lwip_bootstrap(void)
             return;
         }
     }
+#elif defined(MODULE_ESP_ETH)
+    esp_eth_setup(&_esp_eth_dev);
+#ifdef MODULE_LWIP_IPV4
+    if (netif_add(&netif[0], IP4_ADDR_ANY, IP4_ADDR_ANY, IP4_ADDR_ANY,
+                  &_esp_eth_dev, lwip_netdev_init, tcpip_input) == NULL) {
+        DEBUG("Could not add esp_now device\n");
```
Comment (benpicco) [inline pkg/lwip/contrib/lwip.c]: Not `esp_ethernet` ?

**Comment by benpicco** [comment]
Please squash!

**Comment by gschorcht** [comment]
@benpicco Squashed.

**Comment by gschorcht** [comment]
Thanks for reviewing and merging.

**Pull request merged**